### PR TITLE
qscintilla: 2.9.4 -> 2.11.2 and sqlitebrowser: 3.11.2

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

sqlitebrowser needs a newer version of qscintilla to work which is included here.

The qt4 version is broken - I will have to run nix-review on this thing to assess the impact.

Anybody on mac who can help to verify on darwin?

Cc: @Ma27 @matthewbauer 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build libsForQt5.qscintilla libsForQt511.qscintilla minc_widgets octave octaveFull octaveHg openscad python27Packages.ovito python27Packages.qscintilla python27Packages.qscintilla-qt5 python37Packages.ovito python37Packages.qscintilla-qt5 qgis qscintilla sonic-pi sqlitebrowser sqliteman tora tortoisehg

**Comment by Ma27** [comment]
First of all thanks a lot for your work! I just checked out your branch and tested the changes. I noticed the following things:

* `sqlitebrowser` broke with a QT plugin failure at startup. I already pushed a fix. I observed the same issue with `tora`, I'll push a fix for this as well, but rebuilds of it take a bit.
* The qt4 variant of qscintilla currently fails to build.
* I currently don't see, how the `arpack` bump is related to the `qscintilla` update and probably out of scope, but I may be missing something.

Regarding the broken darwin build: I don't own Apple hardware, so I doubt that I'll be helpful there. IMHO it was right to mark the build as broken on darwin for now, I'd keep that as-is unless a darwin user wants to investigate :)

**Comment by Ma27** [comment]
`tora` and `sqlitebrowser` are fixed now btw :)

**Comment by peterhoeg** [comment]
Thanks @Ma27 for feedback *and* the patch. 

The arpack change was due to ```nix review``` failing when testing this but otherwise unrelated.

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build sqlitebrowser

**Comment by FRidh** [comment]
What is the status of this PR?

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build qscintilla libsForQt5.qscintilla

File: pkgs/development/libraries/qscintilla/default.nix [hunk 1]
```diff
@@ -3,28 +3,29 @@
...
   pname = "qscintilla";
-  version = "2.9.4";
+  version = "2.11.1";
 
   name = "${pname}-${if withQt5 then "qt5" else "qt4"}-${version}";
 
   src = fetchurl {
-    url = "mirror://sourceforge/pyqt/QScintilla2/QScintilla-${version}/QScintilla_gpl-${version}.zip";
-    sha256 = "04678skipydx68zf52vznsfmll2v9aahr66g50lcqbr6xsmgr1yi";
+    url = "https://www.riverbankcomputing.com/static/Downloads/QScintilla/QScintilla_gpl-${version}.tar.gz";
```
Comment (c0bw3b) [inline pkgs/development/libraries/qscintilla/default.nix]: Already moved to
`https://www.riverbankcomputing.com/static/Downloads/QScintilla/2.11.1/QScintilla_gpl-2.11.1.tar.gz`

**Comment by peterhoeg** [comment]
I'll run nix-review on this when I get to a slightly less anemic machine...

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build sqlitebrowser

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build sqlitebrowser

**Comment by lsix** [comment]
I have another fix for qscintilla in  #68375 (did not see you did it already). My modifications are slightly different (I keep the `_qt?` suffix, just symling it to `libqscintilla2.so`). I guess both approaches are fine, but one needs to make it to master since the qscintilla update holds on other updates.

**Review by ttuegel — APPROVED** [review]

**Pull request merged**